src/core/connectivity.jl

```diff
@@ -213,20 +220,84 @@ struct LatticeConnectivity <: AbstractConnectivity
 
         @assert +(qubits_per_row...) == qubit_count "Failed to build lattice"
 
-        new(qubits_per_row, (nrows, ncols), excluded_positions)
+        sorted_connections =
+            get_sorted_excluded_connections_for_lattice(nrows, ncols, excluded_connections)
+
+        new(qubits_per_row, (nrows, ncols), excluded_positions, sorted_connections)
     end
 
     function LatticeConnectivity(
         qubits_per_row::Vector{Int},
         excluded_positions = Vector{Int}(),
+        excluded_connections = Vector{Tuple{Int,Int}}(),
     )
         nrows = maximum(qubits_per_row)
-        ncols = ceil(+(qubits_per_row...) / nrows)
+        ncols = ceil(Int, +(qubits_per_row...) / nrows)
+
+        sorted_connections =
+            get_sorted_excluded_connections_for_lattice(nrows, ncols, excluded_connections)
 
-        new(qubits_per_row, (nrows, ncols), excluded_positions)
+        new(qubits_per_row, (nrows, ncols), excluded_positions, sorted_connections)
     end
 end
 
+function get_sorted_excluded_connections_for_lattice(
+    nrows::Int,
+    ncols::Int,
+    excluded_connections::Vector{Tuple{Int,Int}},
+)::Vector{Tuple{Int,Int}}
+
+    num_connections = length(excluded_connections)
+    sorted_connections = Vector{Tuple{Int,Int}}(undef, num_connections)
+    for (i_connection, connection) in enumerate(excluded_connections)
+        if connection[1] == connection[2]
+            throw(AssertionError("connection $connection must connect to different qubits"))
+        end
+
+        if connection[1] > connection[2]
+            sorted_connection = (connection[2], connection[1])
+        else
+            sorted_connection = connection
+        end
+
+        if sorted_connection[1] < 1
+            throw(
+                AssertionError(
+                    "connection $connection must have qubits with indices greater than 0",
+                ),
+            )
+        end
+
+        num_qubits = nrows * ncols
+        if sorted_connection[2] > num_qubits
+            throw(
+                AssertionError(
+                    "connection $connection must have qubits with indices less than" *
+                    " $(num_qubits + 1)",
```

Comment:
theres no +1 here

Reply:
If there are `n` qubits, the largest qubit index is `n`. The qubit indices therefore have to be less than `n + 1`.

Reply:
oh! right